doc/news/_preparation_next_release.md

```diff
@@ -41,7 +41,42 @@ From now on `keyNew ("/array/#10", KEY_END)` will create a `Key` with name `/arr
 The whole internal implementation for `keySetName`, `keyAddName`, etc. has been completely rewritten.
 If you rely on specific behaviour of Elektra's Keynames and have already taken the two changes above into account, please refer to the newly created [Keyname documentation](../keynames) and (easier to read) [Python reference implementation](../keynames/keynames.py).
 
-<!-- TODO (kodebach): finish -->
+**Other keyname-related changes**
+
+- Metakeys now use the namespace `meta:/`.
+  The accessor function `keyGetMeta` and `keySetMeta` automatically add this namespace to preserve compatibility.
+  However, if you use the recently introduced `keyMeta` or otherwise directly access the keyname of a metakey, you will have to update your code.
+
+- There was an internal update to how warnings are generated.
+  For users this means that the `warnings` metadata now forms a proper array.
+  Specifically, the first 100 warnings are stored below to the meta keys `warnings/#0`, `warnings/#1`, ..., `warnings/#9`, `warnings/#_10`, ..., `warnings/#_99`.
+  After that, warnings will wrap around, so the 101st warning will be stored as `warnings/#0`, no. 102 as `warnings/#1` etc.
+
+- `default:/` is a new namespace used for keys that exist purely to represent a default value (e.g. generated by the `spec` plugin).
+
+  Looking up cascading keys with `ksLookup` now looks at namespaces in the following order:
+
+  - `proc:/`
+  - `dir:/`
+  - `user:/`
+  - `system:/`
+  - `default:/`
+  - `/` (cascading key itself)
+
+  The final lookup of the cascading key itself, will be removed in a future release.
```

Comment:
Can you create an issue about that? Or even better: lets drop the support now.

Reply:
I will create an issue. The change should be done before 1.0, but I'm not sure it will be done in time for the next release depending on how much breaks by removing this.

I think there might be some components that use `KeySet`s with only cascading keys as a general map-like data structure. Removing the literal cascading name lookup would break those.

The best way forward would probably be to disallow storing cascading keys in `KeySet` altogether. For the general data structure use case, we could add a flag that switches the `KeySet` to "only cascading keys allowed". A similar flag could be introduced for metadata. To prevent something like `ksAppendKey (keyMeta (k), keyNew (...))` (or the even worse `ksAppendKey (keyMeta(k), k)`).

Reply:
Yes, I actually like the idea to only allow a specific kind of keys (cascading, meta or "normal" KeySets). We forgot to discuss this in our decision sessions...

ksLookup by itself also adds cascading keys to the KeySet (if spec contains default). Did you fix this place already?

Reply:
I assume you are talking about this:

https://github.com/ElektraInitiative/libelektra/blob/d9a8478478b0009971a070d2990af0b310abfd7e/src%2Flibs%2Felektra%2Fkeyset.c#L1671-L1689

If so, yes that has been updated.

Reply:
I added #3597 and #3598